### ISSUE TITLE
Clang-14 crashes during compilation with fsanitize flags

### ISSUE THREAD

**Issue by apach301** [llvm/llvm-project#60442]
HI! I was trying to build PyTorch project with `afl-clang-fast` compiler and get some crashes during compilation of [XNNPACK](https://github.com/google/XNNPACK/tree/ae108ef49aa5623b896fc93d4298c49d1750d9ba) dependency. Using `clang-14` to build project directly doesn't cause any errors.

The crash is reproduced if option `-fsanitize=address,undefined` to afl-clang-fast is specified. During build it calls clang-14 with many sanitizers enabled. While testing generated run-scripts for crash reproducing I found that options `-fsanitize=pointer-overflow` and `-fsanitize-recover=pointer-overflow` triggers the error.
Run-scripts and preprocessed sources: [clang-crash.tar.gz](https://github.com/llvm/llvm-project/files/10557268/clang-crash.tar.gz)

Compilation command failes with message:
```console
fatal error: error in backend: Cannot emit physreg copy instruction
Stack dump:
0.	Program arguments: /usr/bin/clang -Wno-unused-command-line-argument -fexperimental-new-pass-manager -fpass-plugin=/usr/local/bin/../lib/afl/SanitizerCoveragePCGUARD.so -DCPUINFO_SUPPORTED_PLATFORM=1 -DFXDIV_USE_INLINE_ASSEMBLY=0 -DXNN_ENABLE_ASSEMBLY=1 -DXNN_ENABLE_MEMOPT=1 -DXNN_ENABLE_SPARSE=1 -DXNN_LOG_LEVEL=0 -I/pytorch_fuzz/third_party/XNNPACK/include -I/pytorch_fuzz/third_party/XNNPACK/src -I/pytorch_fuzz/third_party/cpuinfo/deps/clog/include -I/pytorch_fuzz/third_party/cpuinfo/include -I/pytorch_fuzz/third_party/pthreadpool/include -I/pytorch_fuzz/third_party/FXdiv/include -I/pytorch_fuzz/third_party/FP16/include -isystem /pytorch_fuzz/third_party/protobuf/src -isystem /pytorch_fuzz/third_party/gemmlowp -isystem /pytorch_fuzz/third_party/neon2sse -g -fsanitize=address,bounds,integer,undefined,null,float-divide-by-zero -O3 -DNDEBUG -fPIC -pthread -std=gnu99 -mavx512f -mavx512cd -mavx512bw -mavx512dq -mavx512vl -O2 -MD -MT confu-deps/XNNPACK/CMakeFiles/XNNPACK.dir/src/qc8-igemm/gen/4x16c8-minmax-fp32-avx512skx.c.o -MF confu-deps/XNNPACK/CMakeFiles/XNNPACK.dir/src/qc8-igemm/gen/4x16c8-minmax-fp32-avx512skx.c.o.d -o confu-deps/XNNPACK/CMakeFiles/XNNPACK.dir/src/qc8-igemm/gen/4x16c8-minmax-fp32-avx512skx.c.o -c /pytorch_fuzz/third_party/XNNPACK/src/qc8-igemm/gen/4x16c8-minmax-fp32-avx512skx.c -g -funroll-loops -D__AFL_HAVE_MANUAL_CONTROL=1 -D__AFL_COMPILER=1 -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION=1 "-D__AFL_FUZZ_INIT()=int __afl_sharedmem_fuzzing = 1;extern unsigned int *__afl_fuzz_len;extern unsigned char *__afl_fuzz_ptr;unsigned char __afl_fuzz_alt[1048576];unsigned char *__afl_fuzz_alt_ptr = __afl_fuzz_alt;" "-D__AFL_COVERAGE()=int __afl_selective_coverage = 1;void __afl_coverage_discard();void __afl_coverage_skip();void __afl_coverage_on();void __afl_coverage_off();" "-D__AFL_COVERAGE_START_OFF()=int __afl_selective_coverage_start_off = 1;" -D__AFL_COVERAGE_ON()=__afl_coverage_on() -D__AFL_COVERAGE_OFF()=__afl_coverage_off() -D__AFL_COVERAGE_DISCARD()=__afl_coverage_discard() -D__AFL_COVERAGE_SKIP()=__afl_coverage_skip() "-D__AFL_FUZZ_TESTCASE_BUF=(__afl_fuzz_ptr ? __afl_fuzz_ptr : __afl_fuzz_alt_ptr)" "-D__AFL_FUZZ_TESTCASE_LEN=(__afl_fuzz_ptr ? *__afl_fuzz_len : (*__afl_fuzz_len = read(0, __afl_fuzz_alt_ptr, 1048576)) == 0xffffffff ? 0 : *__afl_fuzz_len)" "-D__AFL_LOOP(_A)=({ static volatile char *_B __attribute__((used,unused));  _B = (char*)\"##SIG_AFL_PERSISTENT##\"; __attribute__((visibility(\"default\"))) int _L(unsigned int) __asm__(\"__afl_persistent_loop\"); _L(_A); })" "-D__AFL_INIT()=do { static volatile char *_A __attribute__((used,unused));  _A = (char*)\"##SIG_AFL_DEFER_FORKSRV##\"; __attribute__((visibility(\"default\"))) void _I(void) __asm__(\"__afl_manual_init\"); _I(); } while (0)" -Wno-unused-command-line-argument
1.	<eof> parser at end of file
2.	Code generation
3.	Running pass 'Function Pass Manager' on module '/pytorch_fuzz/third_party/XNNPACK/src/qc8-igemm/gen/4x16c8-minmax-fp32-avx512skx.c'.
4.	Running pass 'Post-RA pseudo instruction expansion pass' on function '@xnn_qc8_igemm_minmax_fp32_ukernel_4x16c8__avx512skx'
 #0 0x00007fb33e25c5a3 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/usr/bin/../lib/libLLVM-14git.so+0xa395a3)
 #1 0x00007fb33e25a43e llvm::sys::RunSignalHandlers() (/usr/bin/../lib/libLLVM-14git.so+0xa3743e)
 #2 0x00007fb33e25ba4d llvm::sys::CleanupOnSignal(unsigned long) (/usr/bin/../lib/libLLVM-14git.so+0xa38a4d)
 #3 0x00007fb33e1958fa (anonymous namespace)::CrashRecoveryContextImpl::HandleCrash(int, unsigned long) CrashRecoveryContext.cpp:0:0
 #4 0x00007fb33e19589b (/usr/bin/../lib/libLLVM-14git.so+0x97289b)
 #5 0x00007fb33e2569a7 llvm::sys::Process::Exit(int, bool) (/usr/bin/../lib/libLLVM-14git.so+0xa339a7)
 #6 0x0000000000413482 (/usr/bin/clang+0x413482)
 #7 0x00007fb33e1a4a11 llvm::report_fatal_error(llvm::Twine const&, bool) (/usr/bin/../lib/libLLVM-14git.so+0x981a11)
 #8 0x00007fb33e1a48e8 (/usr/bin/../lib/libLLVM-14git.so+0x9818e8)
 #9 0x00007fb340ae5408 llvm::X86InstrInfo::copyPhysReg(llvm::MachineBasicBlock&, llvm::MachineInstrBundleIterator<llvm::MachineInstr, false>, llvm::DebugLoc const&, llvm::MCRegister, llvm::MCRegister, bool) const (/usr/bin/../lib/libLLVM-14git.so+0x32c2408)
#10 0x00007fb33e4ccbab (anonymous namespace)::ExpandPostRA::runOnMachineFunction(llvm::MachineFunction&) ExpandPostRAPseudos.cpp:0:0
#11 0x00007fb33e5cb2ce llvm::MachineFunctionPass::runOnFunction(llvm::Function&) (/usr/bin/../lib/libLLVM-14git.so+0xda82ce)
#12 0x00007fb33e39654d llvm::FPPassManager::runOnFunction(llvm::Function&) (/usr/bin/../lib/libLLVM-14git.so+0xb7354d)
#13 0x00007fb33e39c2e3 llvm::FPPassManager::runOnModule(llvm::Module&) (/usr/bin/../lib/libLLVM-14git.so+0xb792e3)
#14 0x00007fb33e396b8f llvm::legacy::PassManagerImpl::run(llvm::Module&) (/usr/bin/../lib/libLLVM-14git.so+0xb73b8f)
#15 0x00007fb344cd00be (anonymous namespace)::EmitAssemblyHelper::EmitAssemblyWithNewPassManager(clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) BackendUtil.cpp:0:0
#16 0x00007fb344cca376 clang::EmitBackendOutput(clang::DiagnosticsEngine&, clang::HeaderSearchOptions const&, clang::CodeGenOptions const&, clang::TargetOptions const&, clang::LangOptions const&, llvm::StringRef, llvm::Module*, clang::BackendAction, std::unique_ptr<llvm::raw_pwrite_stream, std::default_delete<llvm::raw_pwrite_stream> >) (/usr/bin/../lib/libclang-cpp.so.14git+0x18c9376)
#17 0x00007fb344fbaf13 clang::BackendConsumer::HandleTranslationUnit(clang::ASTContext&) (/usr/bin/../lib/libclang-cpp.so.14git+0x1bb9f13)
#18 0x00007fb343f60823 clang::ParseAST(clang::Sema&, bool, bool) (/usr/bin/../lib/libclang-cpp.so.14git+0xb5f823)
#19 0x00007fb3457adb07 clang::FrontendAction::Execute() (/usr/bin/../lib/libclang-cpp.so.14git+0x23acb07)
#20 0x00007fb34572d206 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/usr/bin/../lib/libclang-cpp.so.14git+0x232c206)
#21 0x00007fb34581eceb clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/usr/bin/../lib/libclang-cpp.so.14git+0x241dceb)
#22 0x000000000041318e cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/usr/bin/clang+0x41318e)
#23 0x00000000004115b8 ExecuteCC1Tool(llvm::SmallVectorImpl<char const*>&) driver.cpp:0:0
#24 0x00007fb3453bbf32 void llvm::function_ref<void ()>::callback_fn<clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, bool*) const::$_1>(long) Job.cpp:0:0
#25 0x00007fb33e19587d llvm::CrashRecoveryContext::RunSafely(llvm::function_ref<void ()>) (/usr/bin/../lib/libLLVM-14git.so+0x97287d)
#26 0x00007fb3453bba69 clang::driver::CC1Command::Execute(llvm::ArrayRef<llvm::Optional<llvm::StringRef> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, bool*) const (/usr/bin/../lib/libclang-cpp.so.14git+0x1fbaa69)
#27 0x00007fb345387ca4 clang::driver::Compilation::ExecuteCommand(clang::driver::Command const&, clang::driver::Command const*&) const (/usr/bin/../lib/libclang-cpp.so.14git+0x1f86ca4)
#28 0x00007fb345388137 clang::driver::Compilation::ExecuteJobs(clang::driver::JobList const&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) const (/usr/bin/../lib/libclang-cpp.so.14git+0x1f87137)
#29 0x00007fb3453a1f68 clang::driver::Driver::ExecuteCompilation(clang::driver::Compilation&, llvm::SmallVectorImpl<std::pair<int, clang::driver::Command const*> >&) (/usr/bin/../lib/libclang-cpp.so.14git+0x1fa0f68)
#30 0x0000000000410fdd main (/usr/bin/clang+0x410fdd)
#31 0x00007fb33d307083 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24083)
#32 0x000000000040e44e _start (/usr/bin/clang+0x40e44e)
clang-14: error: clang frontend command failed with exit code 70 (use -v to see invocation)
clang version 14.0.0 (https://github.com/llvm/llvm-project.git de5b16d8ca2d14ff0d9b6be9cf40566bc7eb5a01)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
clang-14: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-14: note: diagnostic msg: /tmp/4x16c8-minmax-fp32-avx512skx-b1da82.c
clang-14: note: diagnostic msg: /tmp/4x16c8-minmax-fp32-avx512skx-b1da82.sh
```

**Comment by DimitryAndric** [comment]
This is likely fixed after 14.0 with e5147f82e1cba6791252d8f44c1a014cd9ea7927 ("[X86] Remove __builtin_ia32_pabs intrinsics and use generic __builtin_elementwise_abs"), so can you please try clang 15.0?

**Comment by apach301** [comment]
Checked afl-clang-fast with clang-15 - everything compiles fine, thanks!
